rotkehlchen/tests/api/test_historical_balances.py

```diff
@@ -826,3 +829,96 @@ def test_get_historical_prices_per_asset(
+            'interval': DAY_IN_SECONDS,
+        },
+    )
+    result = assert_proper_sync_response_with_result(response)
+    assert str(future_ts) not in result['prices']
+    assert future_ts not in result['no_prices_timestamps']
+    assert future_ts not in result['rate_limited_prices_timestamps']
+
+    # test excluded timestamps
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'historicalpricesperassetresource',
+        ),
+        json={
+            'asset': 'BTC',
+            'from_timestamp': START_TS,
+            'to_timestamp': START_TS + 2 * DAY_IN_SECONDS,
+            'interval': DAY_IN_SECONDS,
+            'exclude_timestamps': [START_TS + DAY_IN_SECONDS],  # Exclude middle timestamp
+        },
+    )
+    result = assert_proper_sync_response_with_result(response)
+    assert str(START_TS) in result['prices']
+    assert str(START_TS + DAY_IN_SECONDS) not in result['prices']
+    assert str(START_TS + 2 * DAY_IN_SECONDS) in result['prices']
+    assert START_TS + DAY_IN_SECONDS not in result['no_prices_timestamps']
+
+    # Test both rate limited and no price found scenarios
+    original_query_historical_price = PriceHistorian.query_historical_price
+
+    def mock_query_historical_price(from_asset, to_asset, timestamp):
+        # Rate limit for one timestamp
+        if timestamp == START_TS + DAY_IN_SECONDS:
+            raise NoPriceForGivenTimestamp(
+                from_asset=from_asset,
+                to_asset=to_asset,
+                time=timestamp,
+                rate_limited=True,
+            )
+        # No price found for another timestamp
+        elif timestamp == START_TS + 2 * DAY_IN_SECONDS:
+            raise NoPriceForGivenTimestamp(
+                from_asset=from_asset,
+                to_asset=to_asset,
+                time=timestamp,
+                rate_limited=False,
+            )
+        # Normal behavior for other timestamps
+        return original_query_historical_price(from_asset, to_asset, timestamp)
+
+    with patch('rotkehlchen.history.price.PriceHistorian.query_historical_price', side_effect=mock_query_historical_price):  # noqa: E501
+        response = requests.post(
+            api_url_for(
+                rotkehlchen_api_server,
+                'historicalpricesperassetresource',
+            ),
+            json={
+                'asset': 'BTC',
+                'from_timestamp': START_TS,
+                'to_timestamp': START_TS + 3 * DAY_IN_SECONDS,
+                'interval': DAY_IN_SECONDS,
+            },
+        )
+        result = assert_proper_sync_response_with_result(response)
+        # The first timestamp should have a price
+        assert str(START_TS) in result['prices']
+
+        # The second timestamp should be rate limited
+        assert str(START_TS + DAY_IN_SECONDS) not in result['prices']
+        assert START_TS + DAY_IN_SECONDS not in result['no_prices_timestamps']
```

Comment:
maybe create a variable for `tomorrow = START_TS + DAY_IN_SECONDS` since you use it a lot in the test. Actually you have it as future_ts